build.zig

```diff
@@ -48,6 +48,21 @@ pub fn build(b: *zbs.Builder) !void {
 
     const examples = b.option(bool, "examples", "Set to true to build examples") orelse false;
 
+    const version = v: {
+        const version_string = "0.1.0";
+
+        var ret: u8 = undefined;
+        const git_dir = fs.path.join(b.allocator, &[_][]const u8{ b.build_root, "/.git" }) catch unreachable;
+        const git_commit_hash = b.execAllowFail(
+            &[_][]const u8{ "git", "--git-dir", git_dir, "--work-tree", b.build_root, "rev-parse", "--short", "HEAD" },
+            &ret,
+            .Ignore,
+        ) catch break :v version_string;
```

Comment:
I don't think we should return a bare version string without the `-dev` if git does not exist. Instead we should return `0.1.0-dev` currently as river `0.1.0` has not yet been released. Maybe we should do something like this:
1. Have a global `const version = "0.1.0-dev";` at the top of the build.zig
2. In the `build()` function, check if the `-dev` suffix is present. If it is, then try and obtain the git commit hash as you are currently doing. If not, simply use the version string as defined.
3. When river 0.1.0 is released, we set the global version const to `0.1.0` for the commit that gets tagged.
4. In the next commit after the 0.1.0 tag, we set the global version const to `0.2.0-dev`.

What do you think of that approach?

Reply:
Sounds good to me

> Have a global const version = "0.1.0-dev"; at the top of the build.zig

Should I use `river_version` so I can use `const version = :v {}` after?

Reply:
I'd use `version` for the global and do `const full_version = blk: {}` after.